Dockerfile

```diff
@@ -0,0 +1,167 @@
+#############################################################
+#                           PHP-FPM                         #
+#############################################################
+ARG PHP_VERSION=7.4
+ARG NODE_VERSION=14
+ARG NGINX_VERSION=1.21
+
+FROM php:${PHP_VERSION}-fpm-alpine AS server-for-symfony-flex-php
+
+WORKDIR /srv/server-for-symfony-flex
+
+# persistent / runtime deps
+RUN apk add --no-cache \
+                acl \
+                bash \
+                file \
+                gettext \
+                git \
+                mariadb-client \
+                openssh-client \
+                libxml2 \
+                libuuid \
+                bind-tools \
+    ;
+
+ARG XDEBUG_VERSION=3.0.4
+
+RUN set -eux; \
+        apk add --no-cache --virtual .build-deps \
+                $PHPIZE_DEPS \
+                coreutils \
+                freetype-dev \
+                icu-dev \
+                libjpeg-turbo-dev \
+                libpng-dev \
+                libtool \
+                libwebp-dev \
+                libzip-dev \
+                mariadb-dev \
+                zlib-dev \
+                libxml2-dev \
+                util-linux-dev \
+        ; \
+        \
+        docker-php-ext-configure gd --with-jpeg --with-webp --with-freetype; \
+        docker-php-ext-configure zip --with-zip; \
+        docker-php-ext-install -j$(nproc) \
+                exif \
+                gd \
+                intl \
+                pdo_mysql \
+                zip \
+                bcmath \
+                sockets \
+                soap \
+        ; \
+        pecl install xdebug-${XDEBUG_VERSION}; \
+        pecl clear-cache; \
+        docker-php-ext-enable \
+                xdebug \
+        ; \
+        runDeps="$( \
+                scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions \
+                        | tr ',' '\n' \
+                        | sort -u \
+                        | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
+        )"; \
+        apk add --no-cache --virtual .sylius-phpexts-rundeps $runDeps; \
+        \
+        apk del .build-deps
+
+COPY --from=composer:1 /usr/bin/composer /usr/bin/composer
```

Comment:
Question: Why do you rely on composer 1 here? There probably is a reason?

Reply:
I tried using composer version 2, but there was a problem with some dependency. I don't recall exactly what problem with which dependency, I'll check it out and try again later and let you know :)

Reply:
The problem was that there were composer plugins that were locked and depending on the composer api v1.
I have updated the dependencies with the plugins now and it's using composer v2 now.